contracts/access/NonceTracker.sol

```diff
@@ -0,0 +1,54 @@
+pragma solidity ^0.4.24;
+
+
+/**
+ * @title NonceTracker
+ * @author Matt Condon (@shrugs)
+ * @dev A simple way to keep track of nonces and restrict access.
+ * Use the `withAccess` modifier to restrict access by address.
+ * Use the `withMaxAccess` modifier to restrict access by address up to a max amount
+ * For example, withMaxAccess(msg.sender, 1) will only allow once-per-address.
+ * You can also accept nonces from users (as part of a hash you verify).
+ */
+contract NonceTracker {
+  mapping(address => uint256) private nonces;
+
+  modifier withAccess(address _address, uint256 _nonce)
```

Comment:
What's the `_nonce` value supposed to be here? I'm not really sure what the intended usage of this modifier is (there are no docs for it btw).

Reply:
I will add docs :)

But it's intended to "register" your usage of some resource, given a specific nonce. For example, using Bouncer, you would pass a custom nonce to avoid replay attacks and invalidate previous signatures. NonceTracker keeps track of the latest nonces and enforces that they only increase

Reply:
Oh I get it now. A bit convoluted though, isn't it? Since you're free to pass any nonce you wish, I don't really see a use for this other than the Bouncer. It's also wildly different from `withMaxAccess` (and incompatible, using one in a contract will break the other), to the point where I'm not sure if it shouldn't be its own thing (a Bouncer extension to prevent replaying txs).